addons/building_markers/XEH_postInit.sqf

```diff
@@ -2,6 +2,13 @@
 
 if (isServer) then {
     [QGVAR(set), LINKFUNC(set)] call CBA_fnc_addEventHandler;
+
+    addMissionEventHandler ["MarkerDeleted", {
+        params ["_marker"];
+
+        [_marker] call CBA_fnc_removeGlobalEventJIP;
```

Comment:
Some checks need to happen before it gets here (i.e filter out all non-ZEN markers).